packages/blockchain-api/src/helpers/KnownAddressesCache.ts

```diff
@@ -0,0 +1,44 @@
+import { DataSnapshot } from '@firebase/database-types'
+import { database } from '../firebase'
+
+const TAG = 'helpers/KnownAddressesCache'
+
+const ROOT_KEY = 'addressesExtraInfo'
+
+const ON_VALUE_CHANGED = 'value'
+
+class KnownAddressesCache {
+  private knownAddresses: { [address: string]: { name?: string; imageUrl?: string } } = {}
```

Comment:
I was returning `{}` when we don't have any info, so we can unpack like:
```
{name, imageUrl} = getValueFor(address)
```
And get undefined values for `name` and `imageUrl` instead of an error if there is no info available.
I can change that and return `undefined` but then I would need to check if the returning value is valid before unpacking. What do you think is the best approach?

Reply:
Yes what I meant is without the ` | undefined` the lookup could seems it's working, while it would crash at runtime.
Basically makes sure we check the result where used (which you already did, but not because TypeScript warned you).
See [this playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBA1mEB3MBBAJugTgUwhPALhgG8YBtAQ0132OiwEswBzAXWLLEoFscB+elCasA3DEY9KLHAFUsAG0EwGzFjAC+mmAF5SGgLAAoYwDMArmGBRG4GDKgA1SgvM4AYiCwAKatjwQQiIsAJScMNx8yqpiElIy8kpBatokxjAwoJCwAG4ubrrwiCgY-vh4VDQBbMbpMLhQ5lhgMHmuOMYaQA) and [this one](https://www.typescriptlang.org/play?#code/MYewdgzgLgBA1mEB3MBBAJugTgUwhPALhgG8YBtAQ0132OiwEswBzAXWLLEoFscB+elCasA3DEY9KLHAFUsAG0EwGzFjAC+MAD4wArmHQ4AZsxzpNMALykNAWABQj4weBRG4GDKgA1Sgr0cADEQLAAKamw8CCERFgBKThhuPmVVMQkpGXklWLVLEkcYGFBIWAA3f0DreEQUDCj8PCoaaLZHIphcKD0sMBhKgJxHDSA) for the difference.

Reply:
Ohhh, got it! Thanks!